kiva/constants.py

```diff
@@ -41,87 +55,209 @@
+
+FILL = DrawMode.FILL
+EOF_FILL = DrawMode.EOF_FILL
+STROKE = DrawMode.STROKE
+FILL_STROKE = DrawMode.FILL_STROKE
+EOF_FILL_STROKE = DrawMode.EOF_FILL_STROKE
 
 # -----------------------------------------------------------------------------
 # Font Constants
 # -----------------------------------------------------------------------------
 
-NORMAL = 0
-BOLD = 1
-ITALIC = 2
-BOLD_ITALIC = 3
+class FontStyle(IntFlag):
+    """Basic font styles."""
+    NORMAL = 0
+    BOLD = 1
+    ITALIC = 2
+
+NORMAL = FontStyle.NORMAL
+BOLD = FontStyle.BOLD
+ITALIC = FontStyle.ITALIC
+BOLD_ITALIC = FontStyle.BOLD | FontStyle.ITALIC
 
 # convenience sets for styles
 bold_styles = {BOLD, BOLD_ITALIC}
 italic_styles = {ITALIC, BOLD_ITALIC}
 
 # Font families, as defined by the Windows API, and their CSS equivalents
-DEFAULT = 0
-SWISS = 1  # Sans-serif
-ROMAN = 2  # Serif
-MODERN = 3  # Monospace
-DECORATIVE = 4  # Fantasy
-SCRIPT = 5  # Cursive
-TELETYPE = 6
+class FontFamily(IntEnum):
+    """Standard font family names"""
+    DEFAULT = 0
+    SWISS = 1  # Sans-serif
+    ROMAN = 2  # Serif
+    MODERN = 3  # Monospace
+    DECORATIVE = 4  # Fantasy
+    SCRIPT = 5  # Cursive
+    TELETYPE = 6
+
+DEFAULT = FontFamily.DEFAULT
+SWISS = FontFamily.SWISS
+ROMAN = FontFamily.ROMAN
+MODERN = FontFamily.MODERN
+DECORATIVE = FontFamily.DECORATIVE
+SCRIPT = FontFamily.SCRIPT
+TELETYPE = FontFamily.TELETYPE
+
 
 # Font weight constants
-WEIGHT_THIN = 100
-WEIGHT_EXTRALIGHT = 200
-WEIGHT_LIGHT = 300
-WEIGHT_NORMAL = 400
-WEIGHT_MEDIUM = 500
-WEIGHT_SEMIBOLD = 600
-WEIGHT_BOLD = 700
-WEIGHT_EXTRABOLD = 800
-WEIGHT_HEAVY = 900
-WEIGHT_EXTRAHEAVY = 1000
+class FontWeight(IntEnum):
+    """Font weights"""
+    THIN = 100
+    EXTRALIGHT = 200
+    LIGHT = 300
+    NORMAL = 400
+    MEDIUM = 500
+    SEMIBOLD = 600
+    BOLD = 700
+    EXTRABOLD = 800
+    HEAVY = 900
+    EXTRAHEAVY = 1000
+
+WEIGHT_THIN = FontWeight.THIN
+WEIGHT_EXTRALIGHT = FontWeight.EXTRALIGHT
+WEIGHT_LIGHT = FontWeight.LIGHT
+WEIGHT_NORMAL = FontWeight.NORMAL
+WEIGHT_MEDIUM = FontWeight.MEDIUM
+WEIGHT_SEMIBOLD = FontWeight.SEMIBOLD
+WEIGHT_BOLD = FontWeight.BOLD
+WEIGHT_EXTRABOLD = FontWeight.EXTRABOLD
+WEIGHT_HEAVY = FontWeight.HEAVY
+WEIGHT_EXTRAHEAVY = FontWeight.EXTRAHEAVY
 
 # -----------------------------------------------------------------------------
 # Text Drawing Mode Constants
 # -----------------------------------------------------------------------------
 
-TEXT_FILL = 0
-TEXT_STROKE = 1
-TEXT_FILL_STROKE = 2
-TEXT_INVISIBLE = 3
-TEXT_FILL_CLIP = 4
-TEXT_STROKE_CLIP = 5
-TEXT_FILL_STROKE_CLIP = 6
-TEXT_CLIP = 7
-TEXT_OUTLINE = 8
+class TextMode(IntEnum):
+    """Text drawing mode.
+
+    Determines how text is drawn to the screen.  If a CLIP flag is set, the
+    font outline is added to the clipping path. Possible values:
+
+    FILL
+        fill the text
+
+    STROKE
+        paint the outline
+
+    FILL_STROKE
+        fill and outline
+
+    INVISIBLE
+        paint it invisibly ??
```

Comment:
Presumably it has some effect on the clipping path? 🤷

Maybe the only backend which implements this mode is Quartz...